src/Connector.SqlServer/Connector/SqlServerConnector.cs

```diff
@@ -338,10 +373,23 @@ public override async Task<bool> VerifyConnection(ExecutionContext executionCont
             }
         }
 
-        public override async Task StoreData(ExecutionContext executionContext, Guid providerDefinitionId, string containerName, IDictionary<string, object> data)
+        public async Task<string> GetCorrelationId()
+        {
+            return await Task.FromResult(Guid.NewGuid().ToString());
+        }
+
+        public async Task StoreData(ExecutionContext executionContext, Guid providerDefinitionId, string containerName, string correlationId, DateTimeOffset timestamp, VersionChangeType changeType, IDictionary<string, object> data)
         {
             try
             {
+                // If we are in Event Stream mode, append extra fields
+                if (StreamMode == StreamMode.EventStream)
+                {
+                    data.Add(TimestampFieldName, timestamp);
+                    data.Add(ChangeTypeFieldName, changeType);
+                    data.Add(CorrelationIdFieldName, correlationId);
+                }
+
                 if (_bulkSupported)
                 {
```

Comment:
This implementation should stay in the `StoreData` call - we should just be calling that with the updated values.